src/Kanbas/Courses/Quizzes/QuizList.tsx

```diff
@@ -1,5 +1,197 @@
+    //             setError("Failed to fetch quizzes");
+    //             setLoading(false);
+    //             setQuizzes([]);
+    //         }
+    //     };
+
+    //     fetchQuizzes();
+    // }, [courseId]);
+    function getAvailabilityStatus(quiz: Quiz): string {
+      const now = new Date();
+      const availableDate = new Date(quiz.availableDate);
+      const untilDate = new Date(quiz.untilDate);
+      if (now < availableDate) {
+          return "Not available yet";
+      } else if (now > untilDate) {
+          return "Expired";
+      } else {
+          return "Available";
+      }
+  }
+  
+
+    return (
+        <div className='flex-fill' >
+            <h1>Quiz List</h1>
+            <button className="add-quiz-btn">+ Quiz</button>
+            <div className="quiz-list-container">
+                {quizzes.length > 0 ? (
+                    quizzes.map((quiz, index) => (
+                        <div key={index} className="quiz-box">
+                            <Link
+                                to={`${quiz._id}`}
+                                style={{
+                                    textDecoration: "none",
+                                    display: "block",
+                                    flexGrow: 1,
+                                }}
+                            >
+                                <div>
+                                    <div
+                                        className="quiz-title"
+                                        style={{ fontWeight: "bold" }}
+                                    >
+                                        {quiz.title}
+                                    </div>
+                                    <div className="quiz-details">
+                                        <span className="availability-status">
+                                {getAvailabilityStatus(quiz)}
+                            </span> |
+                                        Due {formatDate(new Date(quiz.dueDate))} | Points:{" "}
+                                        {quiz.points} | {quiz.questions.length}{" "}
+                                        Questions
+                                    </div>
+                                </div>
+                            </Link>
+                            <span
+                                className={`publish-status ${
+                                    quiz.isPublished ? "published" : "unpublished"
+                                }`}
+                            >
+                                {quiz.isPublished ?  <FaCheckCircle className='text-success' /> : "🚫"}
+                            </span>
+                            <div
+                                onClick={(e) => e.stopPropagation()}
+                                className="quiz-context-menu"
+                            >
+                                {/* Implement the context menu here */}
+                                ...

```

Comment:
is this also something that you're still working on implementing, or is this being completed as part of someone else's work?